dask/dataframe/io/parquet/arrow.py

```diff
@@ -265,279 +92,674 @@ def _index_in_schema(index, schema):
-                if not col_chunk_paths or (
-                    len(dataset.pieces) != dataset.metadata.num_row_groups
-                ):
-                    dataset.schema = dataset.metadata.schema
-                    dataset.metadata = None
-        else:
-            partitions = []
-
-        # Statistics are currently collected at the row-group level only.
-        # Therefore, we cannot perform filtering with split_row_groups=False.
-        # For "partitioned" datasets, each file (usually) corresponds to a
-        # row-group anyway.
-        # TODO: Map row-group statistics onto file pieces for filtering.
-        #       This shouldn't be difficult if `col_chunk_paths==True`
-        if not split_row_groups and not col_chunk_paths:
-            if gather_statistics is None and not partitions:
-                gather_statistics = False
-                if filters:
-                    raise ValueError(
-                        "Filters not supported with split_row_groups=False "
-                        "(unless proper _metadata is available)."
-                    )
-            if gather_statistics and not partitions:
-                raise ValueError(
-                    "Statistics not supported with split_row_groups=False."
-                    "(unless proper _metadata is available)."
-                )
+    return dataset, base, fns
+
+
+def _get_ds_metadata(ds, ds_filters):
+    metadata = []
+    for file_frag in ds.get_fragments(filter=ds_filters):
+        for rg_frag in file_frag.split_by_row_group():
+            metadata.append(rg_frag)
+    return sorted(metadata, key=lambda x: natural_sort_key(x.path))
 
-        if dataset.metadata:
-            schema = dataset.metadata.schema.to_arrow_schema()
-        else:
-            schema = dataset.schema.to_arrow_schema()
-        columns = None
 
-        has_pandas_metadata = (
-            schema.metadata is not None and b"pandas" in schema.metadata
+def _gather_metadata(
+    paths, fs, split_row_groups, gather_statistics, filters, use_pa_ds, dataset_kwargs
+):
+    """ Gather parquet metadata into a single data structure.
+
+        Use _metadata or aggregate footer metadata into a single
+        object.  Also, collect other information necessary for
+        parquet-to-ddf mapping (e.g. schema, partition_info).
+    """
+
+    if use_pa_ds:
+        # Use pyarrow.dataset API
+        ds = None
+        if len(paths) == 1 and fs.isdir(paths[0]):
+            paths = paths[0]
+
+            # Use _metadata file if available
+            # TODO: Handle _metadata within a list of files
+            meta_path = fs.sep.join([paths, "_metadata"])
+            if fs.exists(meta_path):
+                schema = pa_ds.dataset(
+                    paths,
+                    format="parquet",
+                    partitioning=dataset_kwargs.get(
+                        "partitioning", "hive"
+                    ),  # Assume "hive" by default
+                ).schema
```

Comment:
What's the reason you need to get the schema from here to pass to `parquet_dataset` ? Normally `parquet_dataset` should get the schema from the `_metadata` file fine I think?

Reply:
Was about to eplore this further - right now, the pandas metadata is missing from the schema generated from the _metadata file alone.

Reply:
Ah, that might be a bug, I can certainly imagine we never tested that in this code path.

Reply:
Confirmed this is a bug in the parquet-specific factory, opened https://issues.apache.org/jira/browse/ARROW-9363

Reply:
Fixed on pyarrow master

Reply:
Nice - Thanks for the quick action on that!